File: examples/python/initializing_with_none_without_api.py
import numpy as np
import proxsuite

H = np.array([[65.0, -22.0, -16.0], [-22.0, 14.0, 7.0], [-16.0, 7.0, 5.0]])
g = np.array([-13.0, 15.0, 7.0])
A = None
b = None
C = None
u = None
l = None

results = proxsuite.proxqp.dense.solve(
    H, g, A, b, C, l, u
)  # it is equivalent to do as well proxsuite.proxqp.dense.solve(H, g)
print("optimal x: {}".format(results.x))

File: examples/cpp/initializing_with_none_without_api.cpp
#include <iostream>
#include "proxsuite/proxqp/dense/dense.hpp"
#include <proxsuite/proxqp/utils/random_qp_problems.hpp> // used for generating a random convex qp

using namespace proxsuite::proxqp;
using T = double;
int
main()
{
  dense::isize dim = 10;
  dense::isize n_eq(0);
  dense::isize n_in(0);
  T strong_convexity_factor(0.1);
  T sparsity_factor(0.15);
  // we generate a qp, so the function used from helpers.hpp is
  // in proxqp namespace. The qp is in dense eigen format and
  // you can control its sparsity ratio and strong convexity factor.
  dense::Model<T> qp_random = utils::dense_strongly_convex_qp(
    dim, n_eq, n_in, sparsity_factor, strong_convexity_factor);

  Results<T> results =
    dense::solve<T>(qp_random.H,
                    qp_random.g,
                    qp_random.A,
                    qp_random.b,
                    qp_random.C,
                    qp_random.l,
                    qp_random.u); // initialization with zero shape matrices
  // it is equivalent to do dense::solve<T>(qp_random.H, qp_random.g,
  // nullopt,nullopt,nullopt,nullopt,nullopt);
  //  print an optimal solution x,y and z
  std::cout << "optimal x: " << results.x << std::endl;
  std::cout << "optimal y: " << results.y << std::endl;
  std::cout << "optimal z: " << results.z << std::endl;
}